pkg/katautils/config.go

```diff
@@ -296,8 +296,8 @@ func (h hypervisor) blockDeviceDriver() (string, error) {
 		return defaultBlockDeviceDriver, nil
 	}
 
-	if h.BlockDeviceDriver != vc.VirtioSCSI && h.BlockDeviceDriver != vc.VirtioBlock {
-		return "", fmt.Errorf("Invalid value %s provided for hypervisor block storage driver, can be either %s or %s", h.BlockDeviceDriver, vc.VirtioSCSI, vc.VirtioBlock)
+	if h.BlockDeviceDriver != vc.VirtioSCSI && h.BlockDeviceDriver != vc.VirtioBlock && h.BlockDeviceDriver != vc.VirtioMmio {
+		return "", fmt.Errorf("Invalid value %s provided for hypervisor block storage driver, can be either %s or %s or %s", h.BlockDeviceDriver, vc.VirtioSCSI, vc.VirtioBlock, vc.VirtioMmio)
```

Comment:
This is getting a bit "hairy" :smile:

It's not a succinct or efficient as a `switch`, but something like the following would make it easier to handle the error message:

```go
supportedBlockDrivers := []string{vc.VirtioSCSI, vc.VirtioBlock, vc.VirtioMmio}

found := false

for _, b := range supportedBlockDrivers { 
    if b == h.BlockDeviceDriver { 
        found = true
        break
    } 
} 

if !found {
    return "", fmt.Errorf("Invalid hypervisor block storage driver %v specified (supported drivers: %v)", h.BlockDeviceDriver, supportedBlockDrivers)
}
```

Reply:
Agree. Fixed